src/runtime/contrib/thrust/thrust.cu

```diff
@@ -35,11 +35,12 @@ namespace contrib {
 using namespace runtime;
 
 // Performs sorting along axis -1 and returns both sorted values and indices.
-template<typename DataType, typename IndicesType>
+template<typename DataType, typename IndicesType, typename F>
 void thrust_sort(DLTensor* input,
                  DLTensor* out_values,
                  DLTensor* out_indices,
-                 bool is_ascend) {
+                 bool is_ascend,
+                 const F &get_sort_len) {
```

Comment:
I think `std::function<int(int)>` looks easier to read than introducing typename F.